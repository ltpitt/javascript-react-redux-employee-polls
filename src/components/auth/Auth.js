import React from "react";
import "./Auth.css";
import { connect } from "react-redux";
import { useState } from "react";
import { Navigate } from "react-router-dom";
import { handleLogin } from "../../actions/authedUser";

const Auth = ({ dispatch, loggedIn, authedUser }) => {
  const [username, setUsername] = useState("sarahedo");
  const [password, setPassword] = useState("password123");
  if (loggedIn) {
    const urlParams = new URLSearchParams(window.location.search);
    const redirectUrl = urlParams.get("redirectTo");
    return <Navigate to={redirectUrl ? redirectUrl : "/"} />;
  }

  const handleUsername = (e) => {
    const value = e.target.value;
    setUsername(value);
  };

  const handlePassword = (e) => {
    const value = e.target.value;
    setPassword(value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(handleLogin(username, password));
  };

  return (
    <div className="Auth-form-container">
      <form className="Auth-form" onSubmit={handleSubmit}>
        <div className="Auth-form-content">
          <h3 className="Auth-form-title">Sign In</h3>
          <div className="form-group mt-3">
            <label>Username</label>
            <input
              value={username}
              onChange={handleUsername}
              type="test"
              className="form-control mt-1"
              placeholder="Enter email"
            />
          </div>
          <div className="form-group mt-3">
            <label>Password</label>
            <input
              value={password}
              onChange={handlePassword}
              type="password"
              className="form-control mt-1"
              placeholder="Enter password"
            />
          </div>
          <div className="d-grid gap-2 mt-3">
            <button type="submit" className="btn btn-primary">
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

const mapStateToProps = ({ authedUser, users }) => ({
  loggedIn: !!authedUser,
  authedUser,
  users,
});

export default connect(mapStateToProps)(Auth);
