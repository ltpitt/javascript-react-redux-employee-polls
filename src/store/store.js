import { configureStore } from "@reduxjs/toolkit";
import authedUserReducer from "../reducers/authedUser";
import questionsReducer from "../reducers/questions";
import usersReducer from "../reducers/users";
import logger from "../middleware/logger";
import thunk from "redux-thunk";
import { combineReducers } from "@reduxjs/toolkit";

export const store = configureStore({
  reducer: combineReducers({
    authedUser: authedUserReducer,
    questions: questionsReducer,
    users: usersReducer,
  }),
  middleware: [thunk, logger],
});
