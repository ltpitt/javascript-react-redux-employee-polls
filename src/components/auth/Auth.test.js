import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../store/store";
import { BrowserRouter } from "react-router";
import React from "react";
import Auth from "./Auth";

describe("Login", () => {
  it("should render and match snapshot", () => {
    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Auth />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
