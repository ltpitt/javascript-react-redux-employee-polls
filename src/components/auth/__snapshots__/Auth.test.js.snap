// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Auth-form-container"
      >
        <form
          class="Auth-form"
        >
          <div
            class="Auth-form-content"
          >
            <h3
              class="Auth-form-title"
            >
              Sign In
            </h3>
            <div
              class="form-group mt-3"
            >
              <label>
                Username
              </label>
              <input
                class="form-control mt-1"
                placeholder="Enter email"
                type="test"
                value="sarahedo"
              />
            </div>
            <div
              class="form-group mt-3"
            >
              <label>
                Password
              </label>
              <input
                class="form-control mt-1"
                placeholder="Enter password"
                type="password"
                value="password123"
              />
            </div>
            <div
              class="d-grid gap-2 mt-3"
            >
              <button
                class="btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Auth-form-container"
    >
      <form
        class="Auth-form"
      >
        <div
          class="Auth-form-content"
        >
          <h3
            class="Auth-form-title"
          >
            Sign In
          </h3>
          <div
            class="form-group mt-3"
          >
            <label>
              Username
            </label>
            <input
              class="form-control mt-1"
              placeholder="Enter email"
              type="test"
              value="sarahedo"
            />
          </div>
          <div
            class="form-group mt-3"
          >
            <label>
              Password
            </label>
            <input
              class="form-control mt-1"
              placeholder="Enter password"
              type="password"
              value="password123"
            />
          </div>
          <div
            class="d-grid gap-2 mt-3"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
